wandb/sdk/interface/interface_shared.py

```diff
@@ -544,6 +547,17 @@ def _communicate_sampled_history(
         assert sampled_history_response
         return sampled_history_response
 
+    def _communicate_job_info(
```

Comment:
you don't need it since you are using publish. In general we are trying to get away from communicate and eventual remove it completely.

Reply:
Removed :)

Reply:
Okay I remember why I added this now, func-linux-s_grpc-py39  and some others are failing with:
```
Traceback (most recent call last):
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/wandb/sdk/wandb_run.py", line 2133, in _atexit_cleanup
    self._on_finish()
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/wandb/sdk/wandb_run.py", line 2381, in _on_finish
    job_info_handle = self._backend.interface.deliver_request_job_info()
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/wandb/sdk/interface/interface.py", line 857, in deliver_request_job_info
    return self._deliver_request_job_info(job_link)
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/wandb/sdk/interface/interface_grpc.py", line 453, in _deliver_request_job_info
    job_info_response = self._stub.JobInfo(job_info)
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/grpc/_channel.py", line 1030, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/mnt/ramdisk/.tox/func-grpc-py39/lib/python3.9/site-packages/grpc/_channel.py", line 910, in _end_unary_response_blocking
    raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Exception calling application: 'InterfaceRelay' object has no attribute '_communicate_job_info'"
	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Exception calling application: \'InterfaceRelay\' object has no attribute \'_communicate_job_info\'", grpc_status:2, created_time:"2023-06-29T20:51:13.969197217+00:00"}"
>
INFO: exit= 255
```

Reply:
You need to fix `JobInfo` as well, you are calling communicate there.